File: lib/src/main/java/com/hammwerk/wizardpager/IntegerPage.java
package com.hammwerk.wizardpager;

import android.support.v4.app.Fragment;

import com.hammwerk.wizardpager.core.Page;

public class IntegerPage extends Page<Integer> {
	private final String unit;

	public IntegerPage(String title, String unit) {
		this(title, true, unit);
	}

	public IntegerPage(String title, boolean required, String unit) {
		super(title, required);
		this.unit = unit;
	}

	@Override
	protected Fragment createFragment() {
		return IntegerPageFragment.createInstance(this, unit);
	}
}

File: lib/src/main/kotlin/com/hammwerk/wizardpager/IntegerPage.kt
package com.hammwerk.wizardpager

import com.hammwerk.wizardpager.core.Page

class IntegerPage @JvmOverloads constructor(title: String, required: Boolean = true, val unit: String) :
		Page(title, IntegerPageFragment::class.java.name, required) {
	var value: Int? = null
}
